proxies/openfeature-nodejs/handlers/location.ts

```diff
@@ -0,0 +1,258 @@
+            logs: [], // TODO add logs
+        }
+    } catch (error) {
+        console.error(error)
+        if (isAsync) {
+            ctx.status = 200
+            ctx.body = {
+                asyncError: error.message,
+                stack: error.stack
+            }
+        } else {
+            ctx.status = 200
+            ctx.body = {
+                exception: error.message,
+                stack: error.stack
+            }
+        }
+    }
+}
+
+const getEntityFromLocation = (location: string, data: DataStore) => {
+    const urlParts = location.replace(/^\//, '').split('/')
+    const [locationType, ...locationPath] = urlParts
+
+    if (locationType === 'command') {
+        const [entityType, commandId] = locationPath
+        return data.commandResults[entityType][commandId]
+    } else if (locationType === 'client') {
+        const [clientId] = locationPath
+        return data.clients[clientId]
+    }
+
+    return undefined
+}
+
+const getEntityFromParamType = (type: 'user' | 'event', body: LocationRequestBody) => {
+    if (type === 'user') {
+        return body.user
+    } else if (type === 'event') {
+        return body.event
+    }
+    return undefined
+}
+
+const parseParams = (body: LocationRequestBody, params: Param[], data: DataStore): ParsedParams => {
+    const parsedParams: ParsedParams = []
+    params.forEach((element) => {
+        if (element.callbackURL) {
+            parsedParams.push(new URL(element.callbackURL))
+        } else if (element.type) {
+            parsedParams.push(getEntityFromParamType(element.type, body))
+        } else {
+            parsedParams.push(element.value)
+        }
+    })
+    return parsedParams
+}
+
+const getOpenFeatureVariable = async (
+    openFeatureClient: OFClient,
+    params: any[]
+): Promise<DVCVariable> => {
+    const [user, key, defaultValue, type] = params
+    if (type === 'boolean') {
+        return dvcVariableFromEvaluationDetails(
+            await openFeatureClient.getBooleanDetails(key, defaultValue as boolean, user),
+            defaultValue
+        )
+    } else if (type === 'number') {
+        return dvcVariableFromEvaluationDetails(
+            await openFeatureClient.getNumberDetails(key, defaultValue as number, user),
+            defaultValue
+        )
+    } else if (type === 'JSON') {
+        return dvcVariableFromEvaluationDetails(
+            await openFeatureClient.getObjectDetails(key, defaultValue as JsonValue, user),
+            defaultValue
+        )
+    } else if (type === 'string') {
+        return dvcVariableFromEvaluationDetails(
+            await openFeatureClient.getStringDetails(key, defaultValue as string, user),
+            defaultValue
+        )
+    } else {
+        throw new Error('Invalid default value type')
+    }
+}
+
+/**
+ * Fake DVCVariable Class so that the variable type reporting works correctly
+  */
+class DVCVariable implements DVCVariableInterface {
```

Comment:
this is a bit gross but it works. this is because the NodeJS SDK returns a `DVCVariable` class, but we only expose the interface publically.